packages/test-base/src/androidAndroidTest/kotlin/io/realm/kotlin/test/shared/LinkingObjectsTests.kt

```diff
@@ -0,0 +1,346 @@
+                    this.copyToRealm(Parent(it))
+                }
+
+                child.assertParents(0)
+
+                parents.forEach { parent ->
+                    parent.child = child
+                    parent.childList.add(child)
+                    parent.childList.add(child)
+                    parent.childSet.add(child)
+                    parent.childSet.add(child)
+                }
+
+                assertEquals(5, child.parents.size)
+                assertEquals(10, child.parentsByList.size)
+                assertEquals(5, child.parentsBySet.size)
+            }
+        }
+    }
+
+    @Test
+    fun recursive() {
+        runBlocking {
+            val recursive = realm.write {
+                val recursive = this.copyToRealm(Recursive())
+                recursive.recursiveField = recursive
+                recursive
+            }
+            assertEquals(1, recursive.references.size)
+            assertEquals(recursive.name, recursive.references[0].name)
+        }
+    }
+
+    @Test
+    fun dynamic() {
+        runBlocking {
+            realm.write {
+                val child = this.copyToRealm(Child())
+
+                this.copyToRealm(
+                    Parent().apply
+                    {
+                        this.child = child
+                        this.childSet.add(child)
+                        this.childList.add(child)
+                    }
+                )
+            }
+
+            realm.asDynamicRealm().query(Child::class.simpleName!!).first().find()!!
+                .let { dynamicObject ->
+                    assertEquals(1, dynamicObject.getLinkingObjects(Child::parents.name).size)
+                    assertEquals(1, dynamicObject.getLinkingObjects(Child::parentsByList.name).size)
+                    assertEquals(1, dynamicObject.getLinkingObjects(Child::parentsBySet.name).size)
+                }
+        }
+    }
+
+    @Test
+    fun dynamicMissingProperty_throws() {
+        runBlocking {
+            realm.write {
+                this.copyToRealm(Recursive())
+            }
+            realm.asDynamicRealm().let { dynamicRealm ->
+                val child = dynamicRealm.query("Recursive").first().find()!!
+                assertFailsWithMessage<IllegalArgumentException>("Schema for type 'Recursive' doesn't contain a property named 'missing'") {
+                    child.getLinkingObjects("missing")
+                }
+            }
+        }
+    }
+
+    @Test
+    fun dynamicWrongProperty_throws() {
+        runBlocking {
+            realm.write {
+                this.copyToRealm(Recursive())
+            }
+            realm.asDynamicRealm().let { dynamicRealm ->
+                val child = dynamicRealm.query("Recursive").first().find()!!
+                assertFailsWithMessage<IllegalArgumentException>("Trying to access property 'name' as linking objects but actual schema type is 'class io.realm.kotlin.types.RealmUUID'") {
```

Comment:
```suggestion
                assertFailsWithMessage<IllegalArgumentException>("Trying to access property 'name' as linking objects but actual schema type is 'class io.realm.kotlin.types.RealmUUID'") {
```

Consider changing this to `"Trying to access property 'name' as an object reference but schema type is 'class io.realm.kotlin.types.RealmUUID'"

Reply:
Also, would be nice with tests for `RealmSet<String>` and `RealmList<String>`, just to make sure that collections that can also contain primitives are handled correctly.